agent/lib_guzzle6.c

```diff
@@ -342,14 +342,62 @@ const zend_function_entry nr_guzzle6_requesthandler_functions[]
 
 /* }}} */
 
+static char* nr_guzzle_version(zval* app TSRMLS_DC) {
+  char* retval = NULL;
+  zval* version = NULL;
+  zend_class_entry* ce = NULL;
+
+  if (0 == nr_php_is_zval_valid_object(app)) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application object is invalid",
+                     __func__);
+    return NULL;
+  }
+
+  ce = Z_OBJCE_P(app);
+  if (NULL == ce) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application has NULL class entry",
+                     __func__);
+    return NULL;
+  }
+
+  if (NULL == nr_php_get_class_constant(ce, "VERSION")) {
+    // If full version does not exist, then we will send empty string
+    return " ";
+  } else {
+    version = nr_php_get_class_constant(ce, "VERSION");
+  }
+
+  if (NULL == version) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application does not have VERSION",
+                     __func__);
+    return NULL;
+  }
+
+  if (nr_php_is_zval_valid_string(version)) {
+    retval = nr_strndup(Z_STRVAL_P(version), Z_STRLEN_P(version));
+  } else {
+    nrl_verbosedebug(NRL_FRAMEWORK,
+                     "%s: expected VERSION be a valid string, got type %d",
+                     __func__, Z_TYPE_P(version));
+  }
+
+  nr_php_zval_free(&version);
+  return retval;
+}
+
 NR_PHP_WRAPPER_START(nr_guzzle6_client_construct) {
+  char* version;
   zval* config;
   zend_class_entry* guzzle_client_ce;
   zval* handler_stack;
   zval* middleware = NULL;
   zval* retval;
   zval* this_var = nr_php_scope_get(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
 
+  version = nr_guzzle_version(this_var TSRMLS_CC);
+  // Add php package to transaction
```

Comment:
I think the code will read cleaner if you put a newline before comments like this.  If that seems agreeable please change other code also in the PR to follow this convention.

Reply:
Also when adding code you do not need the `TMRMLS_CC` so remove that from the PR as well for any code you are adding.

Reply:
Fixed: [971ec5a](https://github.com/newrelic/newrelic-php-agent/pull/775/commits/971ec5ae8685f5ee8a978ea1b35001b44712e274)